### DISCUSSION TITLE
Bin file to NES File how?

### DISCUSSION THREAD

**Discussion by Imuseless4ever** [sanni/cartreader#688] [Need help]
Hello, I created an account especially for this question, so I'm not sure if I'm doing everything right.  I've tried one with a nes game, however, the file is a bin, is it possible to output it directly as a nes file and if not, how do you change it?

**Comment by sanni** [comment]
Can you post the last entry of the OSCR_LOG.txt found in the root of the SD card?

Example:
````
OSCR HW5 V12.1

[+] NES/Famicom

Searching database
for 4AD01697 or 433706F0...

NES CART READER
CURRENT SETTINGS
MAPPER:   0
PRG SIZE: 32K
CHR SIZE: 8K
RAM SIZE: 0K

Press Button...

[+] Read iNES Rom

Saving to NES/ROM/SuperMarioBros/6/...
[*******************]
CRC32... 9A2DB086 -> Super Mario Bros. (Europe).nes

Press Button...
````

**Reply by Imuseless4ever** [reply]
Eh...yes but where can I find it?

**Reply by sanni** [reply]
![sdlogfile](https://user-images.githubusercontent.com/19169494/213703072-d056e00f-6a83-42e8-b626-dd485ddb4934.jpg)

If you don't have that file then maybe you have an old version and need to update.

**Reply by Imuseless4ever** [reply]
ah ok but it will take a while until I can send it

**Reply by Imuseless4ever** [reply]
![IMG_20230121_012333438.jpg](https://user-images.githubusercontent.com/123172629/213828633-51527a59-b79a-496e-ad86-75e74daddf4f.jpg)

where is it supposed to be found? I searched and couldn't find anything.  I just sent the card like that

**Reply by sanni** [reply]
You need to update your Cart Reader to the newest software version and also copy new files to the SD: 
https://github.com/sanni/cartreader/wiki/How-to-flash-the-Arduino
https://github.com/sanni/cartreader/releases/tag/V12.0
https://github.com/sanni/cartreader/wiki/Preparing-the-SD-card#copy-the-database-files-to-the-root-of-the-sd-card

**Reply by Imuseless4ever** [reply]
OK is that difficult and can you do a lot wrong?

**Reply by Imuseless4ever** [reply]
because to be honest I read the last one through and I don't think I understand much but I would just give it a try

**Reply by sanni** [reply]
The Open Source Cartridge Reader is basically just an Arduino with a special shield attached. So the difficulty is the same as working with any other Arduino. There are many tutorials about Arduino on YouTube if you want to get a more general overview.

The specific instructions for the OSCR are:

0) Download the latest portable Release, like V12.0_BETA_Portable.zip, and unzip it
1) Install CH341 Drivers by running SETUP.exe in the "CH341 Drivers" folder
2) Launch arduino.exe found in "Arduino IDE Portable" directory
3) In the Arduino IDE go File -> Sketchbook -> Card_Reader
4) Then Tools -> Board and select "Arduino/Genduino Mega or Mega 2560"
5) Followed by Tools -> Port and select your Arduino
6) In Cart_Reader.ino define your HW version by removing // in front of "#define HWX" where X is your hardware version
7) Next Sketch -> Upload
8) Verify that the upload didn't give you any errors
9) Copy the content of the "SD Card" folder to the root of your SD card
10) Mark the *.txt files as hidden

Download portable Release: https://github.com/sanni/cartreader/releases/tag/V12.0
More info on flashing with the Arduino IDE: https://github.com/sanni/cartreader/wiki/How-to-flash-the-Arduino
More info on SD card files: https://github.com/sanni/cartreader/wiki/Preparing-the-SD-card#copy-the-database-files-to-the-root-of-the-sd-card

How to tell what hardware version you got:
![hrehwwggwgw](https://user-images.githubusercontent.com/19169494/214012771-63094a89-a023-44fa-97bc-b5c50d645d96.jpeg)

**Reply by Imuseless4ever** [reply]
sorry for being so bad, thank you very much for your patience
